cmd/ndm_daemonset/probe/usedbyprobe.go

```diff
@@ -200,13 +200,14 @@ func getBlockDeviceZFSPartition(bd blockdevice.BlockDevice) (string, bool) {
 // isBlockDeviceInUseByKernel tries to open the device exclusively to check if the device is
 // being held by some process. eg: If kernel zfs uses the disk, the open will fail
 func isBlockDeviceInUseByKernel(path string) (bool, error) {
-	_, err := os.OpenFile(path, os.O_EXCL, 0444)
+	f, err := os.OpenFile(path, os.O_EXCL, 0444)
 
 	if errors.Is(err, syscall.EBUSY) {
 		return true, nil
 	}
 	if err != nil {
 		return false, err
 	}
+	defer f.Close()
```

Comment:
should there be a nil check for `f` ?

Reply:
`f.Close()` [internally](https://golang.org/src/os/file_unix.go?s=7277:7305#L223) has a nil check.